pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -546,7 +548,11 @@ func (a awsDefaulter) defaultAWS(m *Machine, config *admissionConfig) (bool, []s
 	}
 
 	if providerSpec.InstanceType == "" {
-		providerSpec.InstanceType = defaultAWSInstanceType
+		if runtime.GOARCH == "arm64" {
```

Comment:
is this how we have done the arm detection gating in other places? (just curious if shouldn't be using a flag or something)

Reply:
We don't currently support heterogeneous clusters, so this is how it is often done.  If that ever changes then yeah things like this would need to change, but AFAIK there is no better way as of now.  Suggestions welcome though.

Reply:
no suggestion currently (aside from my hint about a magic flag _somewhere_), i was just curious how it is being handled in other places.